util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -0,0 +1,284 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package worker implements the worker related activities like starting TCP/UDP/HTTP client/server
+// and collecting the metric output to be returned to the controller when requested.
+package worker
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"k8s.io/perf-tests/util-images/network/netperfbenchmark/pkg/util"
+
+	"k8s.io/klog"
+)
+
+// Worker hold data required for facilitating measurement
+type Worker struct {
+	resultStatus chan string
+	result       []string
+	tstDuration  string
+	tstProtocol  string
+	startedAt    int64
+	futureTime   int64
+}
+
+// startResponse is the response sent back to controller on starting a measurement
+type startWorkResponse struct {
+	error string
```

Comment:
Why string?
Why not error?

Reply:
This struct is used only as a http payload and the field would be read by the  controller as response to http request. Since it's not a golang stack we are handling here but a http request response, made it as a string.

Reply:
Modified